spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/TraceFilterTests.java

```diff
@@ -110,6 +110,21 @@ public void startsNewTrace() throws Exception {
 	}
 
 	@Test
+	public void startsNewTraceWithParentIdInHeaders() throws Exception {
+		this.request = builder()
+				.header(Span.SPAN_ID_NAME, Span.toHex(1L))
+				.header(Span.TRACE_ID_NAME, Span.toHex(2L))
+				.header(Span.PARENT_ID_NAME, Span.toHex(3L))
+				.buildRequest(new MockServletContext());
+		TraceFilter filter = new TraceFilter(this.tracer, this.traceKeys);
+
+		filter.doFilter(this.request, this.response, this.filterChain);
```

Comment:
add a comment saying that this creates a child span (which is why we'd expect the parents to include 1L)